cmd/execution/main.go

```diff
@@ -161,6 +176,36 @@ func main() {
 			pendingBlocks = buffer.NewPendingBlocks() // for following main chain consensus
 			return nil
 		}).
+		Component("Block data uploader", func(builder cmd.NodeBuilder, node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			if enableBlockDataUpload {
+
+				logger := node.Logger.With().Str("component_name", "block_data_uploader").Logger()
+				gcpBucketUploader, err := uploader.NewGCPBucketUploader(
+					context.Background(),
+					gcpBucketName,
+					logger,
+				)
+				if err != nil {
+					return nil, fmt.Errorf("cannot create GCP Bucket uploader: %w", err)
+				}
+
+				asyncUploader := uploader.NewAsyncUploader(
+					gcpBucketUploader,
+					logger,
+					collector,
+				)
+
+				blockDataUploader = asyncUploader
+
+				return asyncUploader, nil
+			}
+
+			// Since we don't have conditional component creation, we just use Noop one.
+			// It's functions will be once per startup/shutdown - non-measurable performance penalty
+			// blockDataUploader will stay nil and disable calling uploader at all
+			return &module.NoopReadDoneAware{}, nil
```

Comment:
An alternative to this is to just omit this particular call to `.Component` altogether if `enableblockDataUpload` is false.

Reply:
This would require you to store the NodeBuilder in a variable first.

Reply:
Not really, as flags are parsed only after NodeBuilder is finished configuring and run

Reply:
That's not true, they are already parsed by the time `Initialize()` is finished running.

What I'm suggesting is, you could do

```
builder := cmd.FlowNode(...).ExtraFlags(...).ValidateFlags(...).Initialize(...)
if enableUpload {
    builder.Component(...)
}
// continue building the remaining modules
builder.Module(...).
```

See [this](https://github.com/onflow/flow-go/blob/5f3460d1814c27907e21c5a85ff5a44b83a3046e/cmd/access/main.go#L382-L429) for an example

Reply:
This is not a a best piece of code architecture-wise, as it break builder pattern with back modification of components.
In this PR this would also mean we change configuration of component that has already been initialized. 

Ideally we would separate init process into two objects - configuration builder and than use it to construct Node object, allowing component to have their own list of dependencies for system to initialize, but that's outside of scope of this PR

Reply:
but the component wouldn't be initialized if `enableUpload` is `false`. I think skipping the component completely would be inline with what is being done with other optional components as well.

Reply:
Agreed, but doing this properly would require refactoring. I find this more coherent and performance penalty is non-existing